### PR TITLE
Proofread the SQLServer syntax of trigger, sequence and index

### PR DESCRIPTION
**Pull request by Icesource**
For #6478.

Changes proposed in this pull request:

- add SQLServer grammar create trigger and sequence
- complete the grammatical definition of alter index and create index

@jingshanglu

### CODE REVIEW

**Comment by terrymanu** [comment]
@Icesource   
Hi, could you rename the title to more make sense?

**Comment by Icesource** [comment]
> @Icesource
> Hi, could you rename the title to more make sense?

@terrymanu  I'm sorry, this is my negligence.  I have renamed the PR

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-sqlserver/src/main/antlr4/imports/sqlserver/DDLStatement.g4 [hunk 1]
```diff
@@ -678,3 +694,97 @@ withCommonTableExpr
...
+    | ALLOW_ROW_LOCKS EQ_ (ON | OFF)
+    | ALLOW_PAGE_LOCKS EQ_ (ON | OFF)
+    | OPTIMIZE_FOR_SEQUENTIAL_KEY EQ_ (ON | OFF)
+    | MAXDOP EQ_ expr
+    | DATA_COMPRESSION EQ_ (NONE | ROW | PAGE | COLUMNSTORE | COLUMNSTORE_ARCHIVE) (ON PARTITIONS LP_ partitionNumberRange (COMMA_ partitionNumberRange)*)?
+    ;
+
+partitionNumberRange
+    : partitionNumber (TO partitionNumber)?
+    ;
```
Comment (jingshanglu) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-sqlserver/src/main/antlr4/imports/sqlserver/DDLStatement.g4:770]: @Icesource Can it be changed like this：
```
partitionNumberRange
    : expr (TO expr)?
    ;
```
and delete rule `partitionNumber`

File: shardingsphere-test/shardingsphere-parser-test/src/main/resources/sql/supported/ddl/create-sequence.xml [hunk 2]
```diff
@@ -20,4 +20,12 @@
...
     <sql-case id="create_sequence_with_option" value="CREATE SEQUENCE TEMPORARY seq_option" db-types="PostgreSQL" />
     <sql-case id="create_sequence_complex" value="CREATE SEQUENCE seq_complex MINVALUE 0 MAXVALUE 100 START 0" db-types="PostgreSQL" />
+    <sql-case id="create_sequence_with_all_arguments" value="CREATE SEQUENCE Test.DecSeq
+        AS decimal(3,0)
+        START WITH 125
+        INCREMENT BY 25
+        MINVALUE 100
+        MAXVALUE 200
+        CYCLE
+        CACHE 3;" db-types="SQLServer" />
```
Comment (jingshanglu) [inline shardingsphere-test/shardingsphere-parser-test/src/main/resources/sql/supported/ddl/create-sequence.xml:30]: @Icesource Can you add some test about `trigger` and `create index`?

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-sqlserver/src/main/antlr4/imports/sqlserver/DDLStatement.g4 (hunk 1 above)
Comment (Icesource) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-sqlserver/src/main/antlr4/imports/sqlserver/DDLStatement.g4]: @jingshanglu  done

File: shardingsphere-test/shardingsphere-parser-test/src/main/resources/sql/supported/ddl/create-sequence.xml (hunk 2 above)
Comment (Icesource) [reply shardingsphere-test/shardingsphere-parser-test/src/main/resources/sql/supported/ddl/create-sequence.xml:30]: @jingshanglu  done, I have added the tests in the new commit

**Pull request merged**